Some compilation warnings

Building the master branch gives the following warnings:
```
[31/216] Building C object src/CMakeFiles/supplements.dir/potential/grdseamount.c.o
../../src/potential/grdseamount.c:515:50: warning: unused parameter 'in' [-Wunused-parameter]
GMT_LOCAL double grdseamount_poly_solver (double in[], double f, double v, bool elliptical) {
                                                 ^
../../src/potential/grdseamount.c:515:81: warning: unused parameter 'elliptical' [-Wunused-parameter]
GMT_LOCAL double grdseamount_poly_solver (double in[], double f, double v, bool elliptical) {
                                                                                ^
2 warnings generated.
[43/216] Building C object src/CMakeFiles/supplements.dir/seis/pspolar.c.o
../../src/seis/pspolar.c:253:69: warning: unused variable 'c' [-Wunused-variable]
        char txt_a[GMT_LEN64] = {""}, txt_b[GMT_LEN64] = {""}, *p = NULL, *c = NULL;
                                                                           ^
1 warning generated.
[109/216] Building C object src/CMakeFiles/gmtlib.dir/gmt_io.c.o
../../src/gmt_io.c:4616:27: warning: unused label 'GMT_IS_RELTIME' [-Wunused-label]
                                        case GMT_IS_ABSTIME: GMT_IS_RELTIME:    /* Possibly convert to periodic time */
                                                             ^~~~~~~~~~~~~~~~~~~
1 warning generated.
[133/216] Building C object src/CMakeFiles/gmtlib.dir/gmt_init.c.o
../../src/gmt_init.c:3620:49: warning: unused parameter 'GMT' [-Wunused-parameter]
void gmtlib_set_case_and_kind (struct GMT_CTRL *GMT, char *format, bool *upper_case, unsigned int *flavor) {
                                                ^
1 warning generated.
[136/216] Building C object src/CMakeFiles/gmtlib.dir/gmt_support.c.o
../../src/gmt_support.c:1845:169: warning: unused parameter 'directed' [-Wunused-parameter]
GMT_LOCAL void gmtsupport_line_angle_line (struct GMT_CTRL *GMT, double x[], double y[], uint64_t start, uint64_t stop, double cangle, uint64_t n, int angle_type, bool directed, struct GMT_LABEL *L) {
                                                                                                                                                                        ^
../../src/gmt_support.c:4912:14: warning: unused variable 'buf' [-Wunused-variable]
        struct stat buf;
                    ^
2 warnings generated.
[179/216] Building C object src/CMakeFiles/gmtlib.dir/movie.c.o
../../src/movie.c:1231:10: warning: type specifier missing, defaults to 'int' [-Wimplicit-int]
        static *sys_cmd_nowait[3] = {"bash", "csh", "start /B"}, *sys_cmd_wait[3] = {"bash", "csh", "cmd /C"};
        ~~~~~~  ^
../../src/movie.c:1231:31: warning: incompatible pointer types initializing 'int *' with an expression of type 'char [5]' [-Wincompatible-pointer-types]
        static *sys_cmd_nowait[3] = {"bash", "csh", "start /B"}, *sys_cmd_wait[3] = {"bash", "csh", "cmd /C"};
                                     ^~~~~~
../../src/movie.c:1231:39: warning: incompatible pointer types initializing 'int *' with an expression of type 'char [4]' [-Wincompatible-pointer-types]
        static *sys_cmd_nowait[3] = {"bash", "csh", "start /B"}, *sys_cmd_wait[3] = {"bash", "csh", "cmd /C"};
                                             ^~~~~
../../src/movie.c:1231:46: warning: incompatible pointer types initializing 'int *' with an expression of type 'char [9]' [-Wincompatible-pointer-types]
        static *sys_cmd_nowait[3] = {"bash", "csh", "start /B"}, *sys_cmd_wait[3] = {"bash", "csh", "cmd /C"};
                                                    ^~~~~~~~~~
../../src/movie.c:1231:60: warning: type specifier missing, defaults to 'int' [-Wimplicit-int]
        static *sys_cmd_nowait[3] = {"bash", "csh", "start /B"}, *sys_cmd_wait[3] = {"bash", "csh", "cmd /C"};
        ~~~~~~                                                    ^
../../src/movie.c:1231:79: warning: incompatible pointer types initializing 'int *' with an expression of type 'char [5]' [-Wincompatible-pointer-types]
        static *sys_cmd_nowait[3] = {"bash", "csh", "start /B"}, *sys_cmd_wait[3] = {"bash", "csh", "cmd /C"};
                                                                                     ^~~~~~
../../src/movie.c:1231:87: warning: incompatible pointer types initializing 'int *' with an expression of type 'char [4]' [-Wincompatible-pointer-types]
        static *sys_cmd_nowait[3] = {"bash", "csh", "start /B"}, *sys_cmd_wait[3] = {"bash", "csh", "cmd /C"};
                                                                                             ^~~~~
../../src/movie.c:1231:94: warning: incompatible pointer types initializing 'int *' with an expression of type 'char [7]' [-Wincompatible-pointer-types]
        static *sys_cmd_nowait[3] = {"bash", "csh", "start /B"}, *sys_cmd_wait[3] = {"bash", "csh", "cmd /C"};
                                                                                                    ^~~~~~~~
../../src/movie.c:1554:27: warning: format specifies type 'char *' but the argument has type 'int *' [-Wformat]
                        sprintf (cmd, "%s %s", sys_cmd_wait[Ctrl->In.mode], pre_file);
                                       ~~      ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/usr/include/secure/_stdio.h:47:56: note: expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
../../src/movie.c:1705:27: warning: format specifies type 'char *' but the argument has type 'int *' [-Wformat]
                        sprintf (cmd, "%s %s", sys_cmd_wait[Ctrl->In.mode], post_file);
                                       ~~      ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/usr/include/secure/_stdio.h:47:56: note: expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
../../src/movie.c:2206:32: warning: format specifies type 'char *' but the argument has type 'int *' [-Wformat]
                sprintf (cmd, "%s %s %*.*d", sys_cmd_nowait[Ctrl->In.mode], master_file, precision, precision, Ctrl->M.frame);
                               ~~            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/usr/include/secure/_stdio.h:47:56: note: expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
../../src/movie.c:2358:35: warning: format specifies type 'char *' but the argument has type 'int *' [-Wformat]
                        sprintf (cmd, "%s %s %*.*d &", sys_cmd_nowait[Ctrl->In.mode], script_file, precision, precision, frame);
                                       ~~              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/usr/include/secure/_stdio.h:47:56: note: expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
../../src/movie.c:2511:27: warning: format specifies type 'char *' but the argument has type 'int *' [-Wformat]
                        sprintf (cmd, "%s %s", sys_cmd_nowait[Ctrl->In.mode], cleanup_file);
                                       ~~      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/usr/include/secure/_stdio.h:47:56: note: expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
13 warnings generated.
[181/216] Building C object src/CMakeFiles/gmtlib.dir/psevents.c.o
../../src/psevents.c:243:33: warning: unused variable 'txt_b' [-Wunused-variable]
        char txt_a[GMT_LEN256] = {""}, txt_b[GMT_LEN256] = {""}, txt_c[GMT_LEN256] = {""};
                                       ^
../../src/psevents.c:243:59: warning: unused variable 'txt_c' [-Wunused-variable]
        char txt_a[GMT_LEN256] = {""}, txt_b[GMT_LEN256] = {""}, txt_c[GMT_LEN256] = {""};
                                                                 ^
2 warnings generated.
[184/216] Building C object src/CMakeFiles/gmtlib.dir/pshistogram.c.o
../../src/pshistogram.c:896:22: warning: '&&' within '||' [-Wlogical-op-parentheses]
        if (!Ctrl->I.active && !GMT->common.R.active[RSET] || GMT->common.R.wesn[YLO] == GMT->common.R.wesn[YHI]) automatic = true;
            ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~
../../src/pshistogram.c:896:22: note: place parentheses around the '&&' expression to silence this warning
        if (!Ctrl->I.active && !GMT->common.R.active[RSET] || GMT->common.R.wesn[YLO] == GMT->common.R.wesn[YHI]) automatic = true;
                            ^
            (                                             )
1 warning generated.
```

--- a/src/gmt_init.c
+++ b/src/gmt_init.c
@@ -3618,6 +3618,7 @@ GMT_LOCAL int gmtinit_init_custom_annot (struct GMT_CTRL *GMT, struct GMT_PLOT_A
 }
 
 void gmtlib_set_case_and_kind (struct GMT_CTRL *GMT, char *format, bool *upper_case, unsigned int *flavor) {
+	gmt_M_unused (GMT);
 	/* Examine the format string and determine if we want upper/lower case and what type of abbreviation, if any */
 	*upper_case = false;	*flavor = 0;	/* Initialize */
 	switch (format[0]) {	/* This parameter controls which version of month/day textstrings we use for plotting */

--- a/src/gmt_io.c
+++ b/src/gmt_io.c
@@ -4613,7 +4613,7 @@ int gmtlib_process_binary_input (struct GMT_CTRL *GMT, uint64_t n_read) {
 					case GMT_IS_DIMENSION:	/* Convert to internal inches */
 						GMT->current.io.curr_rec[col_no] *= GMT->session.u2u[GMT->current.setting.proj_length_unit][GMT_INCH];
 						break;
-					case GMT_IS_ABSTIME: GMT_IS_RELTIME:	/* Possibly convert to periodic time */
+					case GMT_IS_ABSTIME: case GMT_IS_RELTIME:	/* Possibly convert to periodic time */
 						if (GMT->current.io.cycle_operator && GMT->current.io.cycle_col == col_no)
 							gmtlib_modulo_time_calculator (GMT, &(GMT->current.io.curr_rec[col_no]));
 					default:	/* Nothing to do unless periodic */

--- a/src/gmt_support.c
+++ b/src/gmt_support.c
@@ -1844,6 +1844,7 @@ GMT_LOCAL void gmtsupport_line_angle_ave (struct GMT_CTRL *GMT, double x[], doub
 /*! . */
 GMT_LOCAL void gmtsupport_line_angle_line (struct GMT_CTRL *GMT, double x[], double y[], uint64_t start, uint64_t stop, double cangle, uint64_t n, int angle_type, bool directed, struct GMT_LABEL *L) {
 	double dx, dy;
+	gmt_M_unused (directed);
 
 	if (start == stop) {	/* Can happen if we want no smoothing but landed exactly on a knot point */
 		if (start > 0)
@@ -4909,7 +4910,6 @@ GMT_LOCAL int gmtsupport_init_custom_symbol (struct GMT_CTRL *GMT, char *in_name
 	char arg[3][GMT_LEN64] = {"", "", ""}, *fill_p = NULL, *pen_p = NULL, *c = NULL;
 	FILE *fp = NULL;
 	struct GMT_CUSTOM_SYMBOL *head = NULL;
-	struct stat buf;
 	struct GMT_CUSTOM_SYMBOL_ITEM *s = NULL, *previous = NULL;
 
 	if ((type = gmt_locate_custom_symbol (GMT, in_name, name, path, &pos)) == 0) return GMT_RUNTIME_ERROR;

--- a/src/movie.c
+++ b/src/movie.c
@@ -1228,7 +1228,7 @@ EXTERN_MSC int GMT_movie (void *V_API, int mode, void *args) {
 	static char *extension[3] = {"sh", "csh", "bat"}, *load[3] = {"source", "source", "call"}, *rmfile[3] = {"rm -f", "rm -f", "del"};
 	static char *rmdir[3] = {"rm -rf", "rm -rf", "rd /s /q"}, *export[3] = {"export ", "setenv ", ""};
 	static char *mvfile[3] = {"mv -f", "mv -f", "move"}, *cpconf[3] = {"\tcp -f %s .\n", "\tcp -f %s .\n", "\tcopy %s .\n"};
-	static *sys_cmd_nowait[3] = {"bash", "csh", "start /B"}, *sys_cmd_wait[3] = {"bash", "csh", "cmd /C"};
+	static char *sys_cmd_nowait[3] = {"bash", "csh", "start /B"}, *sys_cmd_wait[3] = {"bash", "csh", "cmd /C"};
 
 	char init_file[PATH_MAX] = {""}, state_tag[GMT_LEN16] = {""}, state_prefix[GMT_LEN64] = {""}, param_file[PATH_MAX] = {""}, cwd[PATH_MAX] = {""};
 	char pre_file[PATH_MAX] = {""}, post_file[PATH_MAX] = {""}, main_file[PATH_MAX] = {""}, line[PATH_MAX] = {""}, version[GMT_LEN32] = {""};

--- a/src/potential/grdseamount.c
+++ b/src/potential/grdseamount.c
@@ -517,6 +517,7 @@ GMT_LOCAL double grdseamount_poly_solver (double in[], double f, double v, bool 
 	double I1 = TWO_PI * (M_PI * sqrt(3.0) / 3.0 - 1.7);	/* I(1) definite integral */
 	double b = (1.0 - v) * (M_PI * f * f * poly_smt_func (f) - poly_smt_vol (f)) - v * I1;
 	double t = 0.0, phi = 0.0, lhs = 0.0, last_lhs;
+	gmt_M_unused (in);
 	while (lhs >= b) {
     	t += DELTA_PHI;
     	last_lhs = lhs;

--- a/src/psevents.c
+++ b/src/psevents.c
@@ -239,8 +239,7 @@ static int parse (struct GMT_CTRL *GMT, struct PSEVENTS_CTRL *Ctrl, struct GMT_O
 	 */
 
 	unsigned int n_errors = 0, pos, n_col = 3, k = 0, id = 0;
-	char *c = NULL,*t_string = NULL;
-	char txt_a[GMT_LEN256] = {""}, txt_b[GMT_LEN256] = {""}, txt_c[GMT_LEN256] = {""};
+	char *c = NULL, *t_string = NULL, txt_a[GMT_LEN256] = {""};
 	struct GMT_OPTION *opt = NULL;
 
 	for (opt = options; opt; opt = opt->next) {

--- a/src/pshistogram.c
+++ b/src/pshistogram.c
@@ -893,7 +893,7 @@ EXTERN_MSC int GMT_pshistogram (void *V_API, int mode, void *args) {
 	F.extremes = Ctrl->L.mode;
 	F.weights = Ctrl->Z.weights;
 	F.T = &(Ctrl->T.T);
-	if (!Ctrl->I.active && !GMT->common.R.active[RSET] || GMT->common.R.wesn[YLO] == GMT->common.R.wesn[YHI]) automatic = true;
+	if (!Ctrl->I.active && (!GMT->common.R.active[RSET] || GMT->common.R.wesn[YLO] == GMT->common.R.wesn[YHI])) automatic = true;
 	if (GMT->common.R.active[RSET]) {	/* Gave -R which initially defines the bins also */
 		gmt_M_memcpy (F.wesn, GMT->common.R.wesn, 4, double);
 		Ctrl->T.T.min = F.wesn[XLO]; Ctrl->T.T.max = F.wesn[XHI];

--- a/src/seis/pspolar.c
+++ b/src/seis/pspolar.c
@@ -250,7 +250,7 @@ static int parse (struct GMT_CTRL *GMT, struct PSPOLAR_CTRL *Ctrl, struct GMT_OP
 	 */
 
 	unsigned int n_errors = 0, n;
-	char txt_a[GMT_LEN64] = {""}, txt_b[GMT_LEN64] = {""}, *p = NULL, *c = NULL;
+	char txt_a[GMT_LEN64] = {""}, txt_b[GMT_LEN64] = {""}, *p = NULL;
 	struct GMT_OPTION *opt = NULL;
 	struct GMTAPI_CTRL *API = GMT->parent;
 

